### PR TITLE
[36] translate "minutes" and "seconds" in time inputs

### LINKED ISSUE #36: Translate the "minutes" string in debate configuration
In debate configuration the "minutes" string is hard-coded and isn't translated to other languages.

**Proposed changes**
- The string is translated to a current language
- The translation changes from singular to plural to match the number (_1 minute_, _5 minutes_, etc.). It will not be 100% correct in Polish, (Polish cardinal numerals are sometimes in Nominative and sometimes in Genitive, e.g. _2 minuty, 5 minut_) but it will still be an improvement. Besides, most UI designers don't care and just leave it at that

### PR DESCRIPTION
**Pull request by Mateusz-Dobrzynski**
Resolves #36.

I ended up implementing non-standard Polish "minutes" conjugation, as this software is targeted primarily at Polish audience. It is 100% correct only with numbers below 12, because Oxford debate speeches usually don't exceed 5 minutes anyway.

### CODE REVIEW

**Comment by jakubmanczak** [comment]
Now that we actually want to tackle this translation at all, I think we do it right.
The approach I'd favor is moving this to `@lib` and implementing a function that returns the appropriate string - something like `useLocalisedNumerosityNoun` which would take in a number and return the appropriate noun.

File: src/components/TimeInput.tsx [hunk 1]
```diff
@@ -14,8 +40,8 @@ const TimeInput = (props: { time: number; setTime: (time: number) => void }) =>
           -
         </button>
         <div className="border-y-2 border-neutral-800 min-w-32 flex flex-row justify-center p-2">
-          {Math.floor(props.time / 60)}
-          {" minutes"}
+          {minutesCount}
+          {minutesCount == 1 ? ` ${minutesSingular}` : ` ${minutesPlural}`}
```
Comment (jakubmanczak) [inline src/components/TimeInput.tsx:44]: Doing the following would eliminate this logic, and make it reusable.

---
Now that we actually want to tackle this translation at all, I think we do it right.
The approach I'd favor is moving this to @lib and implementing a function that returns the appropriate string - something like useLocalisedNumerosityNoun which would take in a number and return the appropriate noun.

**Review by jakubmanczak — CHANGES_REQUESTED** [review]

**Review by jakubmanczak — CHANGES_REQUESTED** [review]
I suggested some changes, along with a question about the way we intend to handle things: is holding default versions of grammatical numbers as strings that we then either return unchanged or replace in special cases really the way to go? Would it not be better if we stored these default cases in `@/lib/file.ts`, and returned "m" or "s" as units for unsupported ones?

**Comment by Mateusz-Dobrzynski** [comment]
While I agree that supporting more cases and better names are good suggestions, omitting curly brackets is not something I would do. In principle, it makes debugging more difficult.

**Comment by jakubmanczak** [comment]
I agree with the general sentiment that the omission of curlies should be avoided sometimes. Particularly, when the statement you'd wrap is in the next line – that's when you're prone to writing more and messing up. And the things Javascript lets you do without ever writing `{}` is criminal. But no obedience to a semantics rule should be absolute. Here, it makes sense to break it. There's nothing to debug here, the logic is simple, all of it is just returns anyway, none of it is hard to follow, no reason to double the code's size on your screen, lost conciseness, readability and clarity etc.

Anyway, why are we using top level PR comments to discuss a specific suggestion when the reply feature exists?

File: src/data/strings.json [hunk 2]
```diff
@@ -389,5 +389,25 @@
...
+  },
+  "secondsPlural": {
+    "pl": "sekund",
+    "en": "seconds",
+    "de": "Sekunden"
+  },
+  "and": {
+    "pl": "i",
+    "en": "and",
+    "de": "und"
```
[the remark this answers is not in the document]
Comment (Mateusz-Dobrzynski) [reply src/data/strings.json:411]: As long as returning the answer doesn't require creating any additional logic, I'd rather reuse the code we already have. Special cases require new logic (that's why they're special, so I covered them in the library.

File: src/components/TimeInput.tsx [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+import { useLang } from "@/lib/useLang";
+import { useLocalizedGrammaticalNumber } from "@/lib/useLocalizedGrammaticalNumber";
+
+const TimeInput = (props: {
+  time: number;
+  setTime: (time: number) => void;
+}) => {
+  const minutesCount = Math.floor(props.time / 60);
+  const secondsCount = Math.floor(props.time % 60);
+  const minutesCardinalNumeral = useLocalizedGrammaticalNumber(
```
Comment (jakubmanczak) [inline src/components/TimeInput.tsx:12]:
```suggestion
  const minutesGrammaticalNumber = useLocalizedGrammaticalNumber(
```

File: src/components/TimeInput.tsx [hunk 4]
```diff
@@ -14,8 +28,8 @@ const TimeInput = (props: { time: number; setTime: (time: number) => void }) =>
           -
         </button>
         <div className="border-y-2 border-neutral-800 min-w-32 flex flex-row justify-center p-2">
-          {Math.floor(props.time / 60)}
-          {" minutes"}
+          {minutesCount}
+          {`  ${minutesCardinalNumeral}`}
```
Comment (jakubmanczak) [inline src/components/TimeInput.tsx:32]:
```suggestion
          {`  ${minutesGrammaticalNumber}`}
```

File: src/data/strings.json [hunk 5]
```diff
@@ -389,5 +389,27 @@
     "en": "Loading…",
     "pl": "Ładowanie…",
     "de": "Laden…"
+  },
+  "minutesSingular": {
+    "pl": "minuta",
+    "en": "minute",
```
Comment (jakubmanczak) [inline src/data/strings.json:395]:
```suggestion
    "en": "minute",
    "pl": "minuta",
```

File: src/data/strings.json [hunk 6]
```diff
@@ -389,5 +389,27 @@
...
+    "de": "Minute"
+  },
+  "minutesPlural": {
+    "pl": "minuty",
+    "en": "minutes",
```
Comment (jakubmanczak) [inline src/data/strings.json:400]:
```suggestion
    "en": "minutes",
    "pl": "minuty",
```

File: src/data/strings.json [hunk 7]
```diff
@@ -389,5 +389,27 @@
...
+    "de": "Minuten",
+    "es": "min"
+  },
+  "secondsPlural": {
+    "pl": "sekund",
+    "en": "seconds",
```
Comment (jakubmanczak) [inline src/data/strings.json:406]:
```suggestion
    "en": "seconds",
    "pl": "sekund",
```

File: package.json [hunk 8]
```diff
@@ -6,25 +6,40 @@
...
     "postcss": "^8",
     "tailwindcss": "^3.3.0",
     "typescript": "^5"
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e",
+  "moduleDirectories": [
+    "node_modules",
+    "src"
+  ]
```
Comment (jakubmanczak) [inline package.json:44]: Don't know what this is but I know I don't want it here. Revert please.

**Review by jakubmanczak — CHANGES_REQUESTED** [review]

**Review by jakubmanczak — APPROVED** [review]
✅✅🚀😍🔥🔥🔥🔥

**Comment by jakubmanczak** [comment]
oh no package lock conflict
